packages/wasm/crate/src/planner.rs

```diff
@@ -449,6 +450,27 @@ pub async fn plan_transaction(
         }));
     }
 
+    for tpr::ActionDutchAuctionWithdraw { auction_id, seq } in
+        request.dutch_auction_withdraw_actions
+    {
+        let auction_id: AuctionId = auction_id
+            .ok_or_else(|| anyhow!("missing auction ID in Dutch auction withdraw action"))?
+            .try_into()?;
+
+        save_auction_nft_metadata_if_needed(auction_id, &storage, seq).await?;
```

Comment:
can you remind me again we make this call in the planner to save the metadata, if we're already doing this in the block processor?

Reply:
Huh, I actually didn't think of this. But now that you raise it, I think we'll still need to do it this way for these two use cases:

### 1. The user creates the auction via `pcli`, then views it via a Prax-connected dapp like minifront.

In this case, we need to save the metadata in the block processor, so it won't yet exist in IndexedDB.

### 2. The user creates the auction via a Prax-connected dapp like minifront.

In this case, we need to save the metadata during the planning request, or else the auction NFT will show up as "Unknown Asset" in the "Output" action in the txn approval dialog.

---

Fortunately, these are keyed by ID, so there won't be duplicate assets in the database.

Reply:
The planner should not have _any_ write access to view server state, precisely for this reason. The data flow should be that the view server state is only modified during sync, precisely so that it doesn’t matter where the action was performed, because the data in the view server should just be a replica of the chain state. Otherwise we have tricky synchronization problems depending on where a transaction was performed — eg what if the user uses prax on two computers?